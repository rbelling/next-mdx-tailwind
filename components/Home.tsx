'use client';
import Markdown from '@/components/Markdown';
import type { MDXRemoteSerializeResult } from 'next-mdx-remote';
import ArticleIntro from '@/components/ArticleIntro';

const components = {
  ArticleIntro,
};

const Home = (props: MDXRemoteSerializeResult) => {
  const updatedProps: MDXRemoteSerializeResult = {
    ...props,
    scope: {
      ...props.scope,
      ...props.frontmatter,
      date: new Date().toDateString(),
    },
  };

  return <Markdown {...updatedProps} components={components} />;
};

export default Home;
