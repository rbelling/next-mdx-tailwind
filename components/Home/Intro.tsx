export type IntroProps = {
  todaysDate: string;
};

const Intro = (props: IntroProps) => {
  const onClick = () => alert(`Clicked!`);
  return (
    <>
      <p className="text-italic">Today is {props.todaysDate}.</p>
      <br />
      <button
        className="rounded-full bg-orange-400	text-white px-4 py-2"
        onClick={onClick}
      >
        Click me
      </button>
    </>
  );
};

export default Intro;
