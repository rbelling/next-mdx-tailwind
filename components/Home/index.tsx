'use client';
import Markdown from '@/components/Markdown';
import type { MDXRemoteSerializeResult } from 'next-mdx-remote';
import Intro, { IntroProps } from '@/components/Home/Intro';

const components = {
  Intro,
};

const Home = (props: MDXRemoteSerializeResult) => {
  const updatedProps: MDXRemoteSerializeResult<IntroProps> = {
    ...props,
    scope: {
      ...props.scope,
      todaysDate: new Date().toDateString(),
    },
  };

  return (
    <>
      {props.frontmatter?.title && (
        <h1 className="title">{props.frontmatter.title}</h1>
      )}
      <Markdown {...updatedProps} components={components} />
    </>
  );
};

export default Home;
