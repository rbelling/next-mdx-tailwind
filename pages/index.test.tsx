import { render, screen } from '@testing-library/react';
import Home from '@/pages/index';

describe('Home', () => {
  it('renders a heading and matches snapshot', () => {
    const { container } = render(<Home />);

    const heading = screen.getByTestId('heading');

    expect(heading).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
