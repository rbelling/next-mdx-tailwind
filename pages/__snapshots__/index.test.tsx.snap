// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders a heading and matches snapshot 1`] = `
<div>
  <main
    class="min-h-full min-w-full flex flex-col p-4"
  >
    <h1
      class="text-red-500"
      data-testid="heading"
    >
      Hi there!
    </h1>
  </main>
</div>
`;
